bpf/aya-bpf/src/helpers.rs

```diff
@@ -20,6 +54,149 @@ pub unsafe fn bpf_probe_read<T>(src: *const T) -> Result<T, c_long> {
+pub unsafe fn bpf_probe_read_user<T>(src: *const T) -> Result<T, c_long> {
+    let mut v: MaybeUninit<T> = MaybeUninit::uninit();
+    let ret = gen::bpf_probe_read_user(
+        v.as_mut_ptr() as *mut c_void,
+        mem::size_of::<T>() as u32,
+        src as *const c_void,
+    );
+    if ret < 0 {
+        return Err(ret);
+    }
+
+    Ok(v.assume_init())
+}
+
+/// Read bytes stored at the _kernelspace_ pointer `src` and store them as a `T`.
+///
+/// Returns a bitwise copy of `mem::size_of::<T>()` bytes stored at the kernelspace address
+/// `src`. See `bpf_probe_read_user` for  reading userspace memory.
+///
+/// # Examples
+///
+/// ```no_run
+/// # #![allow(dead_code)]
+/// # use aya_bpf::{cty::{c_int, c_long}, helpers::bpf_probe_read_kernel};
+/// # fn try_test() -> Result<(), c_long> {
+/// # let kernel_ptr: *const c_int = 0 as _;
+/// let my_int: c_int = unsafe { bpf_probe_read_kernel(kernel_ptr)? };
+///
+/// // Do something with my_int
+/// # Ok::<(), c_long>(())
+/// # }
+/// ```
+///
+/// # Errors
+///
+/// On failure, this function returns a negative value wrapped in an `Err`.
+#[inline]
+pub unsafe fn bpf_probe_read_kernel<T>(src: *const T) -> Result<T, c_long> {
+    let mut v: MaybeUninit<T> = MaybeUninit::uninit();
+    let ret = gen::bpf_probe_read_kernel(
+        v.as_mut_ptr() as *mut c_void,
+        mem::size_of::<T>() as u32,
+        src as *const c_void,
+    );
+    if ret < 0 {
+        return Err(ret);
+    }
+
+    Ok(v.assume_init())
+}
+
+/// Read a null-terminated string stored at `src` into `dest`.
+///
+/// Generally speaking, the more specific [`bpf_probe_read_user_str`] and
+/// [`bpf_probe_read_kernel_str`] should be preferred over this function.
+///
+/// The length of `dest` should be the same or smaller than the length or `src`.
+/// In case `dest` is smaller, the read bytes will be truncated to the size of `dest`.
+///
+/// # Examples
+///
+/// ```no_run
+/// # #![allow(dead_code)]
+/// # use aya_bpf::{cty::c_long, helpers::bpf_probe_read_str};
+/// # fn try_test() -> Result<(), c_long> {
+/// # let kernel_ptr: *const u8 = 0 as _;
+/// let mut my_str = [0u8; 16];
+/// let num_read = unsafe { bpf_probe_read_str(kernel_ptr, &mut my_str)? };
+///
+/// // Do something with num_read and my_str
+/// # Ok::<(), c_long>(())
+/// # }
+/// ```
+///
+/// # Errors
+///
+/// On failure, this function returns Err(-1).
+#[inline]
+pub unsafe fn bpf_probe_read_str(src: *const u8, dest: &mut [u8]) -> Result<usize, c_long> {
+    let len = gen::bpf_probe_read_str(
+        dest.as_mut_ptr() as *mut c_void,
+        dest.len() as u32,
+        src as *const c_void,
+    );
+    if len < 0 {
+        return Err(-1);
+    }
+
+    let mut len = len as usize;
+    if len > dest.len() {
+        // this can never happen, it's needed to tell the verifier that len is
+        // bounded
+        len = dest.len();
+    }
+    Ok(len as usize)
+}
+
+/// Read a null-terminated from _userspace_ stored at `src` into `dest`.
```

Comment:
missing "string".

Here and in other places, I'd use user space, and kernel space, with a... space :P